pkg/registrar/api.go

```diff
@@ -0,0 +1,66 @@
+package registrar
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/giantswarm/microerror"
+	clouddns "google.golang.org/api/dns/v1"
+	capg "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
+)
+
+const EndpointAPI = "api"
+
+type API struct {
+	baseDomain string
+	dnsService *clouddns.Service
+}
+
+func NewAPI(baseDomain string, dnsService *clouddns.Service) *API {
+	return &API{
+		baseDomain: baseDomain,
+		dnsService: dnsService,
+	}
+}
+
+func (c *API) Register(ctx context.Context, cluster *capg.GCPCluster) error {
+	if cluster.Spec.ControlPlaneEndpoint.Host == "" {
+		return nil
```

Comment:
I think it would be useful to log that we are skipping until the `ControlPlaneEndpoint.Host` field is set. Otherwise the operator does nothing and we don't know what's going on.